tools/flags.cc

```diff
@@ -0,0 +1,322 @@
+                value);
+        return false;
+      }
+      return true;
+    }
+
+    bool ValidateInt32Percent(const char* flagname, int32_t value) {
+      if (value < 0 || value > 100) {
+        fprintf(stderr, "Invalid value for --%s: %d, 0<= pct <=100 \n",
+                flagname, value);
+        return false;
+      }
+      return true;
+    }
+
+    bool ValidateUint32Range(const char* flagname, uint64_t value) {
+      if (value > std::numeric_limits<uint32_t>::max()) {
+        fprintf(stderr, "Invalid value for --%s: %lu, overflow\n", flagname, value);
+        return false;
+      }
+      return true;
+    }
+
+    bool ValidateRateLimit(const char* flagname, double value) {
+      const double EPSILON = 1e-10;
+      if ( value < -EPSILON ) {
+        fprintf(stderr, "Invalid value for --%s: %12.6f, must be >= 0.0\n",
+                flagname, value);
+        return false;
+      }
+      return true;
+    }
+
+    bool ValidatePrefixSize(const char* flagname, int32_t value) {
+      if (value < 0 || value > 2000000000) {
+        fprintf(stderr, "Invalid value for --%s: %d. 0<= PrefixSize <=2000000000\n",
+                flagname, value);
+        return false;
+      }
+      return true;
+    }
+
+    bool ValidateKeySize(const char* /*flagname*/, int32_t /*value*/) {
+        return true;
+    }
+}  // namespace tools_flags
+
+#ifndef ROCKSDB_LITE
+DEFINE_string(env_uri, "",
+              "URI for registry Env lookup. Mutually exclusive with --hdfs.");
+#endif  // ROCKSDB_LITE
+
+DEFINE_string(hdfs, "",
+             "Name of hdfs environment. Mutually exclusive with --env_uri.");
+
+DEFINE_string(db, "", "Use the db with the following name.");
+
+DEFINE_uint64(seed, 0, "Seed base for random number generators. "
+              "When 0 it is deterministic.");
+
+static const bool FLAGS_seed_dummy __attribute__((__unused__)) =
+    RegisterFlagValidator(&FLAGS_seed, &tools_flags::ValidateUint32Range);
+
+DEFINE_string(
+    options_file, "",
+    "The path to a RocksDB options file.  If specified, then the tool will "
+    "run with the RocksDB options in the default column family of the "
+    "specified options file. "
+    "Note that with this setting, db_bench will ONLY accept the following "
+    "RocksDB options related command-line arguments, all other arguments "
+    "that are related to RocksDB options will be ignored:\n"
+    "\t--use_existing_db\n"
+    "\t--use_existing_keys\n"
+    "\t--statistics\n"
+    "\t--row_cache_size\n"
+    "\t--row_cache_numshardbits\n"
+    "\t--enable_io_prio\n"
+    "\t--dump_malloc_stats\n"
+    "\t--num_multi_db\n"
+    "db_stress will IGNORE all the options that passed via the options file.");
+
+DEFINE_int32(threads, 1, "Number of concurrent threads to run.");
+
+DEFINE_bool(disable_wal, false, "If true, do not write WAL for write.");
+
+DEFINE_bool(enable_pipelined_write, true,  // db_stress use false as default
```

Comment:
@siying I re-arrange the common options in `flags.cc`, and as you see the most default values are the same. While several of the them differs, I just comment as above. I don't know how to make it better. Do you have more ideas?

Reply:
Many of our and (possibly) other users' scripts may rely on the previous default value of such flags to function properly. Adding a comment is not sufficient to remind users to check all these flags with default values.